### ISSUE TITLE
Adding subscriber fails

### ISSUE THREAD

**Issue by intelity-abailey** [cachethq/Cachet#944]
```
[2015-09-11 22:54:14] production.ERROR: exception 'ErrorException' with message 'Argument 1 passed 
to CachetHQ\Cachet\Handlers\Events\SendSubscriberVerificationEmailHandler::handle() must be an instance of CachetHQ\Cachet\Events\SubscriberHasSubscribedEvent, instance of CachetHQ\Cachet\Events\Subscriber\SubscriberHasSubscribedEvent given' in /opt/cachet/45ParkLane/app/Handlers/Events/SendSubscriberVerificationEmailHandler.php:46
Stack trace:
#0 /opt/cachet/45ParkLane/app/Handlers/Events/SendSubscriberVerificationEmailHandler.php(46): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(4096, 'Argument 1 pass...', '/opt/cachet/45P...', 46, Array)
#1 [internal function]: CachetHQ\Cachet\Handlers\Events\SendSubscriberVerificationEmailHandler->handle(Object(CachetHQ\Cachet\Events\Subscriber\SubscriberHasSubscribedEvent))
#2 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9506): call_user_func_array(Array, Array)
#3 [internal function]: Illuminate\Events\Dispatcher->Illuminate\Events\{closure}(Object(CachetHQ\Cachet\Events\Subscriber\SubscriberHasSubscribedEvent))
#4 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9451): call_user_func_array(Object(Closure), Array)
#5 /opt/cachet/45ParkLane/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(705): Illuminate\Events\Dispatcher->fire(Object(CachetHQ\Cachet\Events\Subscriber\SubscriberHasSubscribedEvent), Array, false)
#6 /opt/cachet/45ParkLane/app/Handlers/Commands/Subscriber/SubscribeSubscriberCommandHandler.php(38): event(Object(CachetHQ\Cachet\Events\Subscriber\SubscriberHasSubscribedEvent))
#7 [internal function]: CachetHQ\Cachet\Handlers\Commands\Subscriber\SubscribeSubscriberCommandHandler->handle(Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand))
#8 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9141): call_user_func(Array, Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand))
#9 [internal function]: Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}(Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand))
#10 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9287): call_user_func(Object(Closure), Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand))
#11 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand))
#12 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9269): call_user_func(Object(Closure), Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand))
#13 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9142): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#14 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9128): Illuminate\Bus\Dispatcher->dispatchNow(Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand), NULL)
#15 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(5178): Illuminate\Bus\Dispatcher->dispatch(Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand))
#16 /opt/cachet/45ParkLane/app/Http/Controllers/Dashboard/SubscriberController.php(61): CachetHQ\Cachet\Http\Controllers\Dashboard\SubscriberController->dispatch(Object(CachetHQ\Cachet\Commands\Subscriber\SubscribeSubscriberCommand))
#17 [internal function]: CachetHQ\Cachet\Http\Controllers\Dashboard\SubscriberController->createSubscriberAction()
#18 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(8581): call_user_func_array(Array, Array)
#19 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(8650): Illuminate\Routing\Controller->callAction('createSubscribe...', Array)
#20 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(8630): Illuminate\Routing\ControllerDispatcher->call(Object(CachetHQ\Cachet\Http\Controllers\Dashboard\SubscriberController), Object(Illuminate\Routing\Route), 'createSubscribe...')
#21 [internal function]: Illuminate\Routing\ControllerDispatcher->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#22 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9287): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#23 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#24 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9269): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#25 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(8631): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#26 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(8616): Illuminate\Routing\ControllerDispatcher->callWithinStack(Object(CachetHQ\Cachet\Http\Controllers\Dashboard\SubscriberController), Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'createSubscribe...')
#27 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(7591): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'CachetHQ\Cachet...', 'createSubscribe...')
#28 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(7562): Illuminate\Routing\Route->runWithCustomDispatcher(Object(Illuminate\Http\Request))
#29 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(7215): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request))
#30 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#31 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9287): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#32 /opt/cachet/45ParkLane/app/Http/Middleware/Authenticate.php(51): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#33 [internal function]: CachetHQ\Cachet\Http\Middleware\Authenticate->handle(Object(Illuminate\Http\Request), Object(Closure))
#34 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9279): call_user_func_array(Array, Array)
#35 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#36 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9269): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#37 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(7216): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#38 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(7204): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#39 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(7189): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#40 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(2106): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#41 [internal function]: Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#42 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9287): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#43 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(12579): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#44 [internal function]: Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#45 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9279): call_user_func_array(Array, Array)
#46 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(11210): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#47 [internal function]: Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#48 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9279): call_user_func_array(Array, Array)
#49 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(12316): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#50 [internal function]: Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#51 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9279): call_user_func_array(Array, Array)
#52 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(12255): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#53 [internal function]: Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#54 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9279): call_user_func_array(Array, Array)
#55 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(2776): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#56 [internal function]: Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#57 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9279): call_user_func_array(Array, Array)
#58 /opt/cachet/45ParkLane/vendor/fideloper/proxy/src/TrustProxies.php(47): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#59 [internal function]: Fideloper\Proxy\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
#60 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9279): call_user_func_array(Array, Array)
#61 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#62 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(9269): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#63 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(2053): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#64 /opt/cachet/45ParkLane/bootstrap/cache/compiled.php(2036): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#65 /opt/cachet/45ParkLane/public/index.php(54): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#66 {main} {"identification":{"id":"5b584f14-ce2a-4f94-8825-0e2857d3bdeb"}}
```

broken in current master (536ae35)

-- CONFIG
```
APP_ENV=production
APP_DEBUG=false
APP_URL=http://something.here.com
APP_KEY=properlyformattedappkey

DB_DRIVER=mysql
DB_HOST=localhost
DB_DATABASE=database
DB_USERNAME=username
DB_PASSWORD=password

CACHE_DRIVER=file
SESSION_DRIVER=file
QUEUE_DRIVER=database

MAIL_DRIVER=smtp
MAIL_HOST=smtp-relay.gmail.com
MAIL_PORT=587
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ADDRESS=somewhere@something.com
MAIL_NAME=My Mail Name

REDIS_HOST=null
REDIS_DATABASE=null
REDIS_PORT=null
```

**Comment by ApfelUser** [comment]
@intelity-abailey 

Your code looks way more readable if you but it like this in your comment:

```
- > ```

code

- > ```
```

( without the **- >** )

**Comment by jbrooksuk** [comment]
Don't use the `master` branch please, it's not stable. You should be using the latest tagged release 1.2.1

**Comment by jbrooksuk** [comment]
Closed in b0ec13fa139483133ccad7a64556914c5c21be4a

**Comment by joecohens** [comment]
:+1:
